draft-packages/split-button/KaizenDraft/SplitButton/SplitButton.elm

```diff
@@ -0,0 +1,229 @@
+module KaizenDraft.SplitButton.SplitButton exposing (Msg(..), SplitButton, default, href, init, menuItems, subscriptions, update, view)
+
+import Browser.Events
+import CssModules exposing (css)
+import Html exposing (Html, a, button, div, span, text)
+import Html.Attributes exposing (dir, style, tabindex)
+import Html.Events exposing (onClick)
+import Html.Events.Extra exposing (onClickPreventDefaultAndStopPropagation)
+import Html.Extra exposing (static, viewIf)
+import Icon.Icon as Icon
+import Icon.SvgAsset exposing (svgAsset)
+import Json.Decode as Decode exposing (Decoder)
+
+
+
+-- DATA
+
+
+type alias SplitButtonId =
+    String
+
+
+type DropdownMenuVisibility
+    = Open SplitButtonId
+    | Closed
+
+
+type alias MenuItem menuItemMsg =
+    { label : String
+    , action : menuItemMsg
+    }
+
+
+
+-- MODEL
+
+
+type SplitButton
+    = SplitButton
+        { id : SplitButtonId
+        , dropdownMenuVisibility : DropdownMenuVisibility
+        , disabled : Bool
+        }
+
+
+init : SplitButton
+init =
+    SplitButton { id = "", dropdownMenuVisibility = Closed, disabled = False }
+
+
+subscriptions : Sub (Msg menuItemMsg)
+subscriptions =
+    let
+        keyDecoder : Decoder (Msg menuItemMsg)
+        keyDecoder =
+            Decode.map toMsg (Decode.field "key" Decode.string)
+
+        toMsg : String -> Msg menuItemMsg
+        toMsg string =
+            case string of
+                "Escape" ->
+                    CloseDropdownMenu
+
+                _ ->
+                    NoOp
+    in
+    Sub.batch
```

Comment:
These subscriptions will be firing for every onClick and keydown on the page even if the split button is not being used. Shouldn't they only fire when a SplitButtonState is `open`?